app/components/attachment_button.js

```diff
@@ -206,11 +206,11 @@ export default class AttachmentButton extends PureComponent {
         }
     };
 
-    hasPhotoPermission = async () => {
+    hasPhotoPermission = async (source) => {
         if (Platform.OS === 'ios') {
             const {formatMessage} = this.context.intl;
             let permissionRequest;
-            const hasPermissionToStorage = await Permissions.check('photo');
+            const hasPermissionToStorage = await Permissions.check(source || 'photo');
```

Comment:
Are these changes related to switching the order of the buttons? They seem to not be related

Reply:
@hmhealey They are somewhat related, cause without this change the Alert was not even showing